srf_generation/velocity_model_generation/generate_3d_velocity_model_perturbation.py

```diff
@@ -154,13 +154,20 @@ def generate_velocity_model_perturbation_file_from_model(
     pd.DataFrame(
         complete_layer_parameters,
         columns=["index", "nx", "ny", "hh", "nz", "h_corr", "v_corr", "sigma", "seed"],
-    )[["nz", "h_corr", "z_corr", "sigma", "seed"]].to_csv(
+    )[["nz", "h_corr", "v_corr", "sigma", "seed"]].to_csv(
         f"{out_file}.csv", index=False, mode="a"
     )
 
-    layer_info = sorted(
-        Pool(n_processes).starmap(create_perturbated_layer, complete_layer_parameters)
-    )
+    try:
+        layer_info = sorted(
+            Pool(n_processes).starmap(
```

Comment:
Pool should be used as a context manager (or calling close at the end)?
https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool

> Warning
> 
> multiprocessing.pool objects have internal resources that need to be properly managed (like any other resource) by using the pool as a context manager or by calling close() and terminate() manually. Failure to do this can lead to the process hanging on finalization.
> 
> Note that is not correct to rely on the garbage colletor to destroy the pool as CPython does not assure that the finalizer of the pool will be called (see object.__del__() for more information).
>

Reply:
Huh, pools are spicier than I thought.
Will do some more edits to handle this

Reply:
Who actually reads the documentation though?

Reply:
I do, but only to check that I'm not talking nonsense haha